synnax/pkg/distribution/channel/lease_proxy.go

```diff
@@ -141,7 +147,38 @@ func (lp *leaseProxy) createFreeVirtual(
 	if err != nil {
 		return err
 	}
-	if err := gorp.NewCreate[Key, Channel]().Entries(&toCreate).Exec(ctx, tx); err != nil {
+
+	toUpdate, err := lp.maybeRetrieveExistingUpdate(ctx, tx, channels, retrieveIfNameExists)
```

Comment:
Let's sit down and walk through the logic of this function. There was some confusing code I left in here, and I think we can make things clearer together.